bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase1_x/MirroringConnection.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.mirroring.hbase1_x;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.BufferedMutator;
+import org.apache.hadoop.hbase.client.BufferedMutatorParams;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.RegionLocator;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.security.User;
+
+public class MirroringConnection implements Connection {
+  private MirroringConfiguration configuration;
+  private Connection primaryConnection;
+  private Connection secondaryConnection;
+
+  /**
+   * The constructor called from {@link
+   * org.apache.hadoop.hbase.client.ConnectionFactory#createConnection(Configuration)} and in its
+   * many forms via reflection with this specific signature.
+   *
+   * <p>Parameters are passed down to ConnectionFactory#createConnection method, connection errors
+   * are passed back to the user.
+   */
+  public MirroringConnection(Configuration conf, boolean managed, ExecutorService pool, User user)
+      throws IOException {
+    assert !managed; // This is always-false legacy hbase parameter.
```

Comment:
When will it be true, or rather what is the behavior when this is true that we are asserting here? 

Also, should we stick with Preconditions for all validation?

Reply:
In HBase > 1.0 it won't be true afaik, because managed connections were removed from the API. Bigtable HBase client also asserts that managed is false.

> Also, should we stick with Preconditions for all validation?

ok, I'll do in in next PRs.